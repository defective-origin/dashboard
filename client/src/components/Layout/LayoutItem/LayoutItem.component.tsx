import React from 'react'

// ---| core |---
import { cn, react } from 'tools'

// ---| components |---
import Block, { BlockProps } from 'components/Block'

// ---| self |---
import './LayoutItem.module.scss'

export type LayoutItemVariant = 'left-aside' | 'right-aside' | 'header' | 'footer' | 'content'

export type LayoutItemProps = BlockProps & {
  v?: LayoutItemVariant | string
  area?: React.CSSProperties['gridArea']
  row?: number | string
  column?: number | string
  rows?: number | string
  columns?: number | string
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <LayoutItem />
 */
export function LayoutItem(props: LayoutItemProps): JSX.Element | null {
  const {
    row = 'auto',
    column = 'auto',
    rows = 'auto',
    columns = 'auto',
    area,
    v,
    className,
    ...otherProps
  } = props
  const _className = cn('layout-item', v && `layout-item--${v}`, className)
  const style = !v ? { gridArea: area ?? [row, column, rows, columns].join('/') } : {}

  return <Block className={_className} style={style} {...otherProps} />
}

LayoutItem.displayName = 'LayoutItem'

export default react.attachOverrides(LayoutItem, {
  LeftAside: { v: 'left-aside' },
  RightAside: { v: 'right-aside' },
  Footer: { v: 'footer' },
  Header: { v: 'header' },
  Content: { v: 'content' },
})
