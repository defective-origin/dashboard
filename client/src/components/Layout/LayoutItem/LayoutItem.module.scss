@import 'src/theme';

:root {
  // --ui-framework-color: var(--account-framework-color);
}

:global {
  .layout-item {
    // fix scroll https://stackoverflow.com/questions/61959291/how-to-get-scrolling-in-a-css-grid
    // variant 1 - problems: hide overlapping items
    // overflow: hidden;
    // variant 2 [preferable]
    min-height: 0;
    min-width: 0;

    &--left-aside { grid-area: left-aside; }
    &--right-aside { grid-area: right-aside; }
    &--header { grid-area: header; }
    &--footer { grid-area: footer; }
    &--content { grid-area: content; }
  }
}
