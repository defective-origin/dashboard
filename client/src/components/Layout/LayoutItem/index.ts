export { default } from './LayoutItem.component'
export * from './LayoutItem.component'

export * from './LayoutItemUI'
export { default as useLayoutItemUIProps } from './LayoutItemUI'
