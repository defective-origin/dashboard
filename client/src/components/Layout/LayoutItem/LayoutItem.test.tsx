// ---| tests |---
import { render } from 'tests'

// ---| self |---
import LayoutItem from './LayoutItem.component'

describe('[LayoutItem] component', () => {
  it('should render component', () => {
    const container = render(<LayoutItem />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
