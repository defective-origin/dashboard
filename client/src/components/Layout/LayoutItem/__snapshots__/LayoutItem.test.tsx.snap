// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[LayoutItem] component > should render component 1`] = `<DocumentFragment />`;
